browser/components/StrategicTableLinkCell.tsx

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import * as React from 'react';
+import { reqJSON } from '../util/index';
+
+interface Props {
+  id: string;
+  type: string;
+}
+
+interface State {
+  group: any;
+  project: any;
+  type: string;
+}
+
+export default class StrategicTableLinkCell extends React.Component<
+  Props,
+  State
+> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      group: null,
+      project: null,
+      type: this.props.type,
+    };
+  }
+
+  async componentDidMount() {
+    const type = this.props.type;
+    if (type === 'group') {
+      const group = await reqJSON('/api/strategicgroups/' + this.props.id);
+      this.setState({ group: group.group });
+    } else if (type === 'project') {
+      const project = await reqJSON(
+        '/api/projects/' + this.props.id.toString()
+      );
+      this.setState({ project: project.projectID[0] });
```

Comment:
This is a function that was created prior that returns projects by project id in list form. Therefore, since the id is unique, only one project, if any, is returned. To access it in the list, I need to call the first object in the list, which is the specific project. It is used in other parts of the project that I didn't work on, so I didn't change the function itself to move from query to oneOrNone. Should I refactor the function or add in a comment?

Reply:
This is a function that was created prior that returns projects by project id in list form. Therefore projectID is the key in the project json object for the array of projects returned. However, since the id is unique, only one project, if any, is returned. To access it in the list, I need to call the first object in the list, which is the specific project. It is used in other parts of the project that I didn't work on, so I didn't change the function itself to move from query to oneOrNone. Should I refactor the function or add in a comment?

Reply:
I would suggest either going with writing an additional function that only returns one value or a wrapper function that calls the existing function but only returns a single value.

Reply:
I wrote an additional function that only returns one value now.